### PR TITLE
Properly export global variables

### PR DESCRIPTION
**Pull request by SpaceIm**
see https://github.com/idealvin/cocoyaxi/pull/212#issuecomment-1018518341

What is important actually is to export any `extern` variable (global variable).

### CODE REVIEW

File: include/co/co.h [hunk 1]
```diff
@@ -13,6 +13,8 @@
 #include "./co/wait_group.h"
 #include <vector>
 
+DEC_bool(disable_co_exit);
+
```
Comment (SpaceIm) [inline include/co/co.h:17]: @idealvin Tell me if you want to isolate this declaration elsewhere.

Comment (idealvin) [reply include/co/co.h:17]: Is `FLG_disable_co_exit` visiable in shared library?

Comment (SpaceIm) [reply include/co/co.h]: yes, see next commit for another solution which keep FLG_disable_co_exit declaration in DEF_main

Comment (idealvin) [reply include/co/co.h]: > yes, see next commit for another solution which keep FLG_disable_co_exit declaration in DEF_main

That's nice! So the `extern` variable will be automatically exported in shared library?

Comment (idealvin) [reply include/co/co.h]: If so, we'd better rename `__codec` to `__coapi`

Comment (SpaceIm) [reply include/co/co.h]: Yes, usually I use something like `<LIBNAME>_API` or `<LIBNAME>_EXPORT` (the later is the default definition generated by a helper function of CMake).

Comment (idealvin) [reply include/co/co.h]: So we can use `__coapi` internally to save some time

Comment (SpaceIm) [reply include/co/co.h]: You can keep `__coapi`, it's fine.

Comment (idealvin) [reply include/co/co.h]: ok

Comment (SpaceIm) [reply include/co/co.h]: Now, do you really want to export `FLG_co_sched_num`, FLG_co_stack_size` & `FLG_co_debug_log`? Their declaration is isolated in a private header, so basically it's not part of public interface.

Comment (idealvin) [reply include/co/co.h]: I didn't remember I have exported these flags somewhere?

Comment (SpaceIm) [reply include/co/co.h]: At least it was the intend, they are declared and defined the same way.

Comment (SpaceIm) [reply include/co/co.h]: It's better to keep the number of global variables as low as possible, so if there is no usage for cocoyaxi users, I would advice to keep them internal.

Comment (idealvin) [reply include/co/co.h]: ok

Comment (SpaceIm) [reply include/co/co.h]: > No need to export them. But they may be used in other .cc file in cocoyaxi.

They are not (for the 3 ones I've listed), their usage is isolated in `scheduler.cc` compilation unit. They don't even need to be declared in `scheduler.h`.

Comment (SpaceIm) [reply include/co/co.h]: One solution:
- extract `__codec extern` from `_CO_DEC_FLAG` & `DEC_string` macro
- sparingly add `__codec extern` in front of `DEC_` in headers where it's needed.

Comment (SpaceIm) [reply include/co/co.h]: And `__codec extern` must not be removed from `DEC_` still it's used by consumers.

**Pull request merged**